.github/workflows/deploy.yml

```diff
@@ -0,0 +1,32 @@
+name: Deploy built master on gh-pages
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  nikola_build:
+    runs-on: ubuntu-latest
+    name: 'Build pages'
+    steps:
+    - name: Check out
+      uses: actions/checkout@v2
+    - name: Setup Python
+      uses: actions/setup-python@v2
+    - name: Install dependencies
+      run: python -m pip install -r requirements.txt
+    - name: Build
+      run: nikola build
+    - name: Deploy
+      if: github.ref == 'refs/heads/master'
+      shell: bash
+      run: |
+          set -eou pipefail
+          git config --global user.name 'GithubAction'
+          git config --global user.email 'githubaction@invalid'
+          git pull --all
+          git remote add deploy https://${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY.git
+          nikola github_deploy -m 'Nikola auto deploy [ci skip]'
```

Comment:
How about using https://github.com/marketplace/actions/deploy-nikola-to-github-pages